src/core/ext/filters/client_channel/client_channel.cc

```diff
@@ -4397,120 +4817,88 @@ const char* PickResultTypeName(
   GPR_UNREACHABLE_CODE(return "UNKNOWN");
 }
 
-void CallData::PickSubchannel(void* arg, grpc_error* error) {
-  grpc_call_element* elem = static_cast<grpc_call_element*>(arg);
-  CallData* calld = static_cast<CallData*>(elem->call_data);
-  ChannelData* chand = static_cast<ChannelData*>(elem->channel_data);
+void LoadBalancedCall::PickSubchannel(void* arg, grpc_error* error) {
+  auto* self = static_cast<LoadBalancedCall*>(arg);
   bool pick_complete;
   {
-    MutexLock lock(chand->data_plane_mu());
-    pick_complete = calld->PickSubchannelLocked(elem, &error);
+    MutexLock lock(self->chand_->data_plane_mu());
+    pick_complete = self->PickSubchannelLocked(&error);
   }
   if (pick_complete) {
-    PickDone(elem, error);
+    PickDone(self, error);
     GRPC_ERROR_UNREF(error);
   }
 }
 
-bool CallData::PickSubchannelLocked(grpc_call_element* elem,
-                                    grpc_error** error) {
-  ChannelData* chand = static_cast<ChannelData*>(elem->channel_data);
+bool LoadBalancedCall::PickSubchannelLocked(grpc_error** error) {
   GPR_ASSERT(connected_subchannel_ == nullptr);
   GPR_ASSERT(subchannel_call_ == nullptr);
-  // If this is a retry, use the send_initial_metadata payload that
-  // we've cached; otherwise, use the pending batch.  The
-  // send_initial_metadata batch will be the first pending batch in the
-  // list, as set by GetBatchIndex() above.
-  // TODO(roth): What if the LB policy needs to add something to the
-  // call's initial metadata, and then there's a retry?  We don't want
-  // the new metadata to be added twice.  We might need to somehow
-  // allocate the subchannel batch earlier so that we can give the
-  // subchannel's copy of the metadata batch (which is copied for each
-  // attempt) to the LB policy instead the one from the parent channel.
+  // Grab initial metadata.
+  auto& send_initial_metadata =
```

Comment:
Can we get a non-auto decl here? As well as a const for clarity? (Since it's only being used on the RHS anyway)